src/bson.c

```diff
@@ -1007,6 +1007,7 @@ PHP_FUNCTION(toArray)
 
 	MAKE_STD_ZVAL(state.zchild);
 	if (!bson_to_zval((const unsigned char *)data, data_len, &state)) {
+		zval_ptr_dtor(&state.zchild);
```

Comment:
This shouldn't be needed.
What if it failed with "Could not initialize BSON iterator"? Then you are dtoring uninitialized value.